velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -275,4 +275,21 @@ struct DateAddFunction {
     result = addToDate(date, DateTimeUnit::kDay, value);
   }
 };
+
+template <typename T>
+struct DateDiffFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      int32_t& result,
+      const arg_type<Date>& date1,
+      const arg_type<Date>& date2) {
+    int64_t value = diffDate(DateTimeUnit::kDay, date2, date1);
+    if (value != (int32_t)value) {
+      VELOX_UNSUPPORTED("integer overflow");
```

Comment:
add a test that hits this case and ensures an exception is raised?

Reply:
Sorry, I found this case can not be raised. Because function `diffTimestamp` [here](https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/DateTimeImpl.h#L189-L193) is not support overflow nowadays. Maybe we can enable this case later.

Reply:
If this is not possible, then let's remove the check.